### DISCUSSION TITLE
Connect to SSH Host -> No Matching Results

### DISCUSSION THREAD

**Discussion by stephen-hill** [lapce/lapce#3230] [Q&A] [answered]
Hello

I have installed the latest version of Lapce v0.4.0 (windows) and I want to connect to a remote SSH server. 

First, I have my .ssh/config setup with the hosts I want to use.

```ssh
Host alpha
	User stephen
	Hostname 192.168.0.2
	Port 1984
	IdentityFile "C:\Users\stephen\.ssh\id_rsa"
	
Host beta
    ProxyJump alpha
    User stephen
    HostName 192.168.0.3
    Port 22
    IdentityFile "C:\Users\stephen\.ssh\id_rsa"
```

Then I click on "Connect to SSH Host". The search box opens and says "No Matching Results"

![image](https://github.com/lapce/lapce/assets/205619/c157a22c-6579-48f1-a5aa-324fdb9c2aad)

No matter what I type in the box, nothing happens. If I press enter, it sort-of connects to nothing.

![image](https://github.com/lapce/lapce/assets/205619/908865f1-9e75-4cb0-ab14-a2046f85c1ea)

Is there something I am doing wrong?

Many Thanks
Stephen

**Comment by dzhou121** [comment]
Does it not take any input when you type? How about clicking on the text input to make it "focus" and type again? 

You should be able to type in `alpha` and enter to connect to alpha.

**Reply by stephen-hill** [reply]
Hi @dzhou121 

Nothing happens at all.

![image](https://github.com/lapce/lapce/assets/205619/cb23652b-6a93-4ac6-af28-030a5323d2e8)

I'm going to make an assumption that either A) Lapse is not loading my ssh config, maybe is looking elsewhere, or B) is not decoding it correctly.

Is there a debug log I can view that might display any related error messages?

Cheers
Stephen

**Reply by panekj** [reply]
> I'm going to make an assumption that either A) Lapse is not loading my ssh config, maybe is looking elsewhere, or B) is not decoding it correctly.

No, Lapce doesn't load the config at all. It's only used when connecting to the host because `ssh` executable will read it.
You have to type username and/or hostname manually and confirm it via `enter`

**Comment by stephen-hill** [comment]
@dzhou121 @panekj 

Thank you for the info.

I've found and looked in the log file and found the following:

```
2024-04-30T09:52:42.126786Z DEBUG lapce_app::proxy::remote: linux armv6l
2024-04-30T09:52:42.126810Z ERROR lapce_app::proxy::remote: detected remote host: linux/UnknownArch
2024-04-30T09:52:42.126835Z ERROR lapce_app::proxy: Failed to start SSH remote: Unknown OS and/or architecture
```

The host is a raspberry PI 1 running Debian 11 (bullseye).

**Reply by panekj** [reply, marked as answer]
32-bit ARM architectures are not supported

**Reply by stephen-hill** [reply]
Thank you for your help.
